### Earth Analytics Bootcamp - Final Project
**Author:** Mark Brown
**Date:** 06/01/2019 

In [1]:
# import nessary python packages
import os
import numpy as np
import pandas as pd
import urllib.request
import matplotlib.pyplot as plt

print("Packages successfully imported")

Packages successfully imported


#### Section 1: Questions 1-10 Using Pandas Dataframes

#### Get Data

Use .urllib.request to download the following .csv file of fires in California and import the data to a pandas dataframe:

1. CA_fires_1992_2015_gt_100_acres.csv from https://ndownloader.figshare.com/files/12835340

The data contains one record for every fire greater than 100 acres that occurred between 1992 and 2015. The dataset has columns for the size of the fire (acres) and for the year and month of the fire, along with other details about the cause, reporting agency, county name, etc.

In [16]:
# change working directory
os.chdir("/Users/markb/earth-analytics-bootcamp/")

# print the current directory
print("The current working directory is:", os.getcwd())

# print space between print statments
print("")

# download the required datset
urllib.request.urlretrieve(url = "https://ndownloader.figshare.com/files/12835340", 
                           filename = "data/CA_fires_1992_2015_gt_100_acres.csv")

# print statment that file succesfully downloaded
print("The file has been successfully downloaded and can be seen in the directory list below:")

# print space between print statments
print("")

# list files in 'data' directory to show that files have successfully downloaded
os.listdir("/Users/markb/earth-analytics-bootcamp/data/")

The current working directory is: C:\Users\markb\earth-analytics-bootcamp

The file has been successfully downloaded and can be seen in the directory list below:



['avg-monthly-precip.txt',
 'avg-precip-months-seasons.csv',
 'avg-temp-months-seasons.csv',
 'boulder-precip-1996-to-2006-months.csv',
 'boulder-precip-2007-to-2017-months-seasons.csv',
 'boulder-temp-2004-to-2009.csv',
 'boulder-temp-2010-to-2014.csv',
 'boulder-temp-2015.txt',
 'boulder-temp-2016.txt',
 'boulder-temp-2017.txt',
 'CA_fires_1992_2015_gt_100_acres.csv',
 'monthly-precip-1988-to-1992.csv',
 'monthly-precip-1993-to-1997.csv',
 'monthly-precip-2002-2013.csv',
 'monthly-precip-mm-1998-to-2017.csv',
 'monthly-temp-2010-to-2013.csv',
 'monthly-temp-2014-to-2017.csv',
 'monthly-temp-cel-1998-to-2017.csv',
 'months.txt',
 'precip-2002-2013-months-seasons.csv',
 'precip_2002_2013.csv',
 'snow-2007-to-2017-months-seasons.csv',
 'snow-2007-to-2017.csv',
 'temp-1991-to-1995-months.csv',
 'temp-1996-to-2000-months.csv']

#### Question 1: Explore Structure of the Pandas Dataframe (2 pts)

Use the appropriate functions to print the first few rows of the pandas dataframe and the last few rows of the dataframe.

Note: as this dataframe contains many records, it is not helpful to print the whole dataframe.

In [18]:
# create new pandas dataframe from the downloaded data

CA_fires_1992_2015_gt_100_acres = pd.read_csv("/Users/markb/earth-analytics-bootcamp/data/CA_fires_1992_2015_gt_100_acres.csv")

# use .head function to print the first few rows of the pandas 
# dataframe

CA_fires_1992_2015_gt_100_acres.head()

fd_unq_id  source_reporting_unit_name                  fire_name  year  \
0    1338131              Mendocino Unit  VANN                       1992   
1     216388           Yuma Field Office                    WALTERS  1992   
2     218766  California Desert District                       MESA  1992   
3    1373316   CDF - San Bernardino Unit  COLLINS                    1992   
4    1373321   CDF - San Bernardino Unit  COVINGTON                  1992   

      month  month_num           cause  fire_size fire_size_class state county  
0  February          2   Equipment Use      120.0               D    CA    NaN  
1     March          3  Debris Burning     1800.0               F    CA    NaN  
2     April          4   Equipment Use     4200.0               F    CA    NaN  
3     April          4           Arson      125.0               D    CA    NaN  
4     April          4           Arson      104.0               D    CA    NaN

In [19]:
CA_fires_1992_2015_gt_100_acres.tail()

fd_unq_id               source_reporting_unit_name fire_name  year  \
4096  300308392                               Butte Unit  RICHVALE  2015   
4097  300308084           CDF - San Benito-Monterey Unit   CIENEGA  2015   
4098  300209443  Sequoia And Kings Canyon National Parks     BURNT  2015   
4099  300293910                    Colorado River Agency   HIGHWAY  2015   
4100  300293894           Ventura County Fire Department   SOLIMAR  2015   

         month  month_num              cause  fire_size fire_size_class state  \
4096   October         10      Equipment Use      250.0               D    CA   
4097   October         10      Miscellaneous      690.0               E    CA   
4098   October         10          Lightning      161.0               D    CA   
4099  November         11  Missing/Undefined      323.0               E    CA   
4100  December         12  Missing/Undefined     1288.0               F    CA   

          county  
4096       Butte  
4097  San Benito  
4098         NaN  
4099   Riverside  
4100     Ventura

#### Question 2: Summarize Fire Size (4 pts)

Use the appropriate function to calculate summary statistics of only the fire size (acres).

In your Markdown documentation for this question, write a sentence or two stating:

* the mean, minimum, and maximum fire size (acres) in this dataset.
* the total number of fires in this dataset.

Hints:

* It can helpful to determine how to select the data you need first before summarizing it.
* You can also review how to run summary statistics on a specific column in a pandas dataframe.

In [24]:
# select data from 'fire size' column and save to new dataframe

CA_fires_1992_2015_fire_size = CA_fires_1992_2015_gt_100_acres[["fire_size"]]

# run summary statistics on new dataframe
CA_fires_1992_2015_fire_size.describe()

fire_size
count    4101.000000
mean     2995.314133
std     13481.045403
min       100.100000
25%       180.000000
50%       354.000000
75%      1155.000000
max    315578.800000

* The mean recorded fire size in California from 1992 to 2015 was 2995.3 acres.
* The minimum recorded fire size greater than 100 acres in California from 1992 to 2015 was 100.1 acres
* The maximum recorded fire size in California from 1992 to 2015 was 315578.8 acres
* The total number of records in this dataset is 4101


#### Question 3: Calculate Total Number of Fires For Each Year (4 pts)

Use the appropriate function to calculate the total number of fires per year, and save as a new dataframe.

Note: the displayed data below only shows the first few rows in the dataset.

Hints:

* Review the use of groupby to run statistics on pandas dataframes.
* Think about what value you want to use to group the data and what value you want to use to determine the total number of fires.

In [32]:
# create a new dataframe from groupby calculation of .count on year
CA_fires_per_year = CA_fires_1992_2015_gt_100_acres.groupby(["year"])[["fd_unq_id"]].count()

# print 'CA_fires_per_year'. Use .head to display first few rows only
CA_fires_per_year.head()

fd_unq_id
year           
1992        237
1993        187
1994        201
1995        189
1996        333

#### Question 4: Reset Index (2 pts)

Use the appropriate function to reset the index of the dataframe created in the previous question, so that the year returns to being a column. Save the reset dataframe as a new dataframe.

Note: the displayed data below only shows the first few rows in the dataset.

In [34]:
# create new dataframe and reset index of month_temp_2010_to_2017_count
CA_fires_per_year_index_reset = CA_fires_per_year.reset_index()

# print new dataframe. Use .head to print only the first few rows
CA_fires_per_year_index_reset.head()

year  fd_unq_id
0  1992        237
1  1993        187
2  1994        201
3  1995        189
4  1996        333